File: query_primer/query_primer__accounting.cs
//何故か AutoFlush = true かつクラスを使わないとエラーになる
using System.Collections.Generic;
using System.Linq;
using static System.Console;

internal class Program
{
    class Purchase
    {
        public string Id { get; set; }
        public string Price { get; set; }

        public Purchase( string p, string m )
        {
            Id = p;
            Price = m;
        }
    }

    static void Main()
    {
        int[] n = ReadLine().Split( ' ' ).Select( int.Parse ).ToArray();
        var d = new Dictionary<string, List<Purchase>>( n[0] );

        foreach( int i in Enumerable.Range( 0, n[0] ) ) {
            d.Add( ReadLine(), new List<Purchase>() );
        }
        foreach( int i in Enumerable.Range( 0, n[1] ) ) {
            string[] s = ReadLine().Split( ' ' );

            d[s[0]].Add( new Purchase( s[1], s[2] ) );
        }
        foreach( KeyValuePair<string, List<Purchase>> p in d ) {
            WriteLine( p.Key );
            foreach( Purchase v in p.Value ) {
                WriteLine( $"{v.Id} {v.Price}" );
            }
            WriteLine( "-----" );
        }
        return;
    }
}

File: query_primer/query_primer__accounting.py
n, k = list(map(int, input().split()))
d = {}

for _ in range(n):
    d[input()] = {}
for _ in range(k):
    a, p, m = input().split()

    if p not in d[a]:
        d[a][p] = int(m)
    else:
        d[a][p] += int(m)
for k1, v1 in d.items():
    print(k1)
    for k2, v2 in d[k1].items():
        print(k2, v2)
    print("-----")
# 実行時間 410ms
